app/controllers/concerns/shopify_app/ensure_installed.rb

```diff
@@ -58,5 +63,12 @@ def shop_login
 
       url.to_s
     end
+
+    def validate_emebedded_session_is_active
+      client = ShopifyAPI::Clients::Rest::Admin.new(session: installed_shop_session)
+      client.get(path: "shop")
+    rescue ShopifyAPI::Errors::HttpResponseError
```

Comment:
I wish the error coming back from API were more descriptive, but I get why they do it. Shouldn't give bad guys clues to bad session states. It does make rescuing from this and handling it more delicate.

I'm assuming we wouldn't want to drop the shop record from storage here? 🤔

Reply:
We could further test this for a `401` code, which would indicate the token is invalid.

Reply:
I would consider keeping the app_uninstalled webhook around for actually deleting stuff from the db to minimize the amount of work we do here / make sure we get all sessions for a shop.